pkg/oss/coco.go

```
@@ -0,0 +1,137 @@
+package oss
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/kubernetes-sigs/alibaba-cloud-csi-driver/pkg/utils/kata/directvolume"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	optDirectAssigned  = "direct"
+	optAnnotations     = "annotations"
+	optEncrypted       = "encrypted"
+	optEncPasswd       = "encPasswd"
+	optKmsKeyId        = "kmsKeyId"
+	fsTypeSecureMount  = "secure_mount"
+	volumeTypeOSS      = "alibaba-cloud-oss"
+	sealedSecretPrefix = "sealed."
+)
+
+func (ns *nodeServer) publishDirectVolume(ctx context.Context, req *csi.NodePublishVolumeRequest, opt *Options) (*csi.NodePublishVolumeResponse, error) {
+	logger := log.WithFields(map[string]interface{}{
+		"NodeServer": "OSSNodeServer",
+		"VolumeId":   req.VolumeId,
+		"TargetPath": req.TargetPath,
+	})
+
+	volumePath := req.TargetPath
+	if isDirectVolumePath(volumePath) {
+		logger.Infof("NodePublishVolume: The mount info for DirectVolume is already exist: %s", volumePath)
+		return &csi.NodePublishVolumeResponse{}, nil
+	}
+
+	device := req.TargetPath
+	volumeType := volumeTypeOSS
+	fsType := fsTypeSecureMount
+	var encrypted, encPasswd, kmsKeyId string
+	annotationsObj := map[string]string{}
+
+	if v := req.Secrets[optEncPasswd]; v != "" {
+		encPasswd = v
+	}
+	for key, value := range req.VolumeContext {
+		switch key {
+		case optEncrypted:
+			encrypted = strings.TrimSpace(value)
+		case optEncPasswd:
+			encPasswd = strings.TrimSpace(value)
+		case optKmsKeyId:
+			kmsKeyId = strings.TrimSpace(value)
+		case optAnnotations:
+			json.Unmarshal([]byte(value), &annotationsObj)
```

Is it intentional to ignore the errors here?